osi_vehicle.proto

```diff
@@ -0,0 +1,692 @@
+
+    //
+    // \brief Interface to the vehicle-model.
+    // The focus here is on the physical description of a wheel.
+    //
+    message Wheel
+    {
+        // Contains the friction-coefficient of each wheel.
+        // Dimensionless.
+        //
+        // Unit: []
+        //
+        optional double friction_coefficient = 1;
+
+        // Contains the x, y and z-coordinate of the contact point of 
+        // the wheel, so that walking, torsion and deflation can be visualized.
+        // relative to the center of the wheel.
+        //
+        // Unit: [m]
+        //
+        optional Vector3d contact_point = 2;
+
+        // Contains the rotational speed of each wheel per second.
+        //
+        // Unit: [rad/s]
+        //
+        optional double rotational_speed = 3;
+
+        // Contains the steering angle of each wheel.
+        //
+        // Unit: [rad]
+        //
+        optional double steeringangle = 4;
+
+        // Contains the camber of each wheel.
+        // 
+        // Negative camber if the bottom of the wheel is farther out than the
+        // top.
+        // For more information: https://en.wikipedia.org/wiki/Camber_angle.
+        //
+        // Unit: [rad]
+        //
+        optional double camber = 5;
+
+        // Contains the tirepressure of each tire.
+        //
+        // Unit: [Pa]
+        // 
+        optional double tirepressure = 6;
+
+        // Contains the springdeflection in z-direction for each wheel.
+        //
+        // Unit: [mm]
+        //
+        optional double springdeflection = 7;
+
+        //Contains the relativ position of the wheel to the center of the car
+        //
+        //Unit: [m]
+        //
+        optional Vector3d position = 8;
+    
+        //Contains the relativ orientation of the wheel to the center of the car
+        //As the rotation of the wheel is also controlled by this value, 
+        //
+        //Unit: [rad]
+        //
+        optional Orientation3d orientation = 9;
+
+        //Contains the absolute (longitunal) slip of the tire
+        //0-100 percent
+        //
+        //Unit: []
+        //
+        optional double slip = 10;
+
+        //Contains the slip-angle of the tire
+        //
+        //Unit: [rad]
+        //
+        optional double slipangle = 11;
+    }
+
+    //
+    // \brief Information about the automated driving system.
+    // Important is the differentiation between states and intentions.
+    // An automated-driving function itself does not have many states. It is more about Intentions,
+    // to change some vehicle states. E.g. a desired acceleration(-pedal-position), but effective is the one in
+    // the powertrain message.
+    //
+    message VehicleAutomatedDrivingFunction
+    {
+        //
+        // Here the name of the function can be chosen.
+        // E.g.: Hadpilot, longitudinal_control, lateral_control or emergency_brake_assistant.
+        //
+        optional string function_name = 1;
```

Comment:
hm... A string is always the easy solution... "Carlos super duper function" <- valid standard name according to OSI

Reply:
A enum with the examples as enumeration would be better right?

Reply:
Done.